### DISCUSSION TITLE
Unable to load file URL

### DISCUSSION THREAD

**Discussion by leonlwwang** [pocketbase/pocketbase#1323] [Q&A] [answered]
Hi!

I'm trying to load a home page icon onto my website's header in Svelte. The icon is stored in my pocketbase collection called `content`.

```svelte
<script>
    import PocketBase from 'pocketbase';
    const pb = new PocketBase('http://127.0.0.1:8090');

    async function getImg() {
        const record = await pb.collection('content').getOne('ael1ai2ju014k72');
        const url = pb.getFileUrl(record, record.file);
        return url;
    }
    let promise = getImg();
</script>
```

But when I try to inject the file url into the header, it gets stuck on `loading...` when I view the live page. Snippet:

```svelte
{#await promise}
    <p>loading...</p>
{:then url}
    <img src={url} alt="">
{:catch error}
    <p>{error}</p>
{/await}
```

I hardcoded the file url directly into the `src` tag and the icon does load. Any ideas on what the issue is here? 

Appreciate the help!

**Comment by ganigeorgiev** [comment, marked as answer]
Is there any error in the browser console or just the promise never resolve?
If you try to `console.log(url)` inside getImg() does it ever get called?

**Reply by leonlwwang** [reply]
Yeah, the file url is revealed when I 'console.log(url)' inside of the function. I wonder if this is a frontend issue instead?

**Reply by ganigeorgiev** [reply]
If `getOne` resolves successfully and the generated url is correct, then most likely it is something frontend related. Based only on the above 2 snippets I think it should work but maybe there is something more in the code that conflicts with the icon loading.
